checkmarx/util.py

```diff
@@ -255,67 +283,66 @@ def scan_gh_artifact(
         exclude_regexes=exclude_paths,
     )
     return upload_and_scan_gh_artifact(
-        artifact_name=scan_artifact.name,
+        artifact=scan_artifact,
         cx_project=cx_project,
         gh_repo=gh_repo,
         source_commit_hash=commit_hash,
         path_filter_func=path_filter_func,
+        force=force,
     )
 
 
-def send_mail(
-    email_recipients,
-    routes: checkmarx.client.CheckmarxRoutes,
-    scans: model.FinishedScans,
-    threshold: int,
-):
-    body = checkmarx.tablefmt.assemble_mail_body(
-        failed_artifacts=scans.failed_scans,
-        routes=routes,
-        scans_above_threshold=scans.scans_above_threshold,
-        scans_below_threshold=scans.scans_below_threshold,
-        threshold=threshold,
-    )
-    try:
-        # get standard cfg set for email cfg
-        default_cfg_set_name = ci.util.current_config_set_name()
-        cfg_factory = ci.util.ctx().cfg_factory()
-        cfg_set = cfg_factory.cfg_set(default_cfg_set_name)
-
-        logger.info(f'sending notification emails to: {",".join(email_recipients)}')
-        mailutil._send_mail(
-            email_cfg=cfg_set.email(),
-            recipients=email_recipients,
-            mail_template=body,
-            subject='[Action Required] checkmarx vulnerability report',
-            mimetype='html',
-        )
-        logger.info('sent notification emails to: ' + ','.join(email_recipients))
-
-    except Exception:
-        traceback.print_exc()
-        logger.warning('error whilst trying to send notification-mail')
+def greatest_severity(result: model.ScanResult) -> model.Severity | None:
+    if result.scan_statistic.highSeverity > 0:
+        return model.Severity.HIGH
+    elif result.scan_statistic.mediumSeverity > 0:
+        return model.Severity.MEDIUM
+    elif result.scan_statistic.lowSeverity > 0:
+        return model.Severity.LOW
+    elif result.scan_statistic.infoSeverity > 0:
+        return model.Severity.INFO
+    else:
+        return None
```

Comment:
If we're just interested in the first one that is `> 0`, we can use
```suggestion
    if result.scan_statistic.highSeverity:
        return model.Severity.HIGH
    elif result.scan_statistic.mediumSeverity:
        return model.Severity.MEDIUM
    elif result.scan_statistic.lowSeverity:
        return model.Severity.LOW
    elif result.scan_statistic.infoSeverity:
        return model.Severity.INFO
    else:
        return None
```
Up to you.

Reply:
Find this a bit confusing for integers in this case. Would be wrong outcome for negative numbers (which should not happen here).